src/components/dialog-full-screen/dialog-full-screen.test.js

```diff
@@ -0,0 +1,355 @@
+      }
+    );
+
+    it("should render DialogFullScreen with disabledEscKey prop and not be closed after clicking Escape button", () => {
+      CypressMountWithProviders(<DialogFullScreenComponent disableEscKey />);
+
+      dialogFullScreenPreview().should("exist");
+      dialogFullScreenPreview().get("body").type("{esc}");
+      cy.wait(1000);
+      dialogFullScreenPreview().should("exist");
+    });
+
+    it("should close DialogFullScreen after pressing Escape button", () => {
+      CypressMountWithProviders(<DialogFullScreenComponent />);
+      dialogFullScreenPreview().should("exist");
+      dialogFullScreenPreview().get("body").type("{esc}");
+      cy.wait(1000);
+      dialogFullScreenPreview().should("not.exist");
+    });
+
+    it("should call the cancel action after closing the DialogFullScreen", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(
+        <DialogFullScreenComponent onCancel={callback} />
+      );
+
+      closeIconButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should allow to close nested DialogFullScreen and then the main DialogFullScreen window", () => {
+      CypressMountWithProviders(<NestedDialog title="title" />);
+
+      openMainDialogButton().click();
+      cy.wait(500);
+
+      openNestedDialogButton().click();
+      cy.wait(500);
+
+      dialogPreview().should("exist");
+
+      portal().contains("Nested Dialog").should("be.visible");
+
+      dialogFullScreenPreview()
+        .contains("Main Dialog")
+        .should("not.be.visible");
+
+      portal().contains("Nested Dialog").get("body").type("{esc}");
+      cy.wait(1000);
+
+      dialogPreview().should("not.exist");
+
+      dialogFullScreenPreview().contains("Main Dialog").should("be.visible");
+
+      dialogFullScreenPreview().get("body").type("{esc}");
+      cy.wait(1000);
+
+      dialogFullScreenPreview().should("not.exist");
+    });
+
+    it("should render DialogFullScreen component with aria-describedby", () => {
+      CypressMountWithProviders(
+        <DialogFullScreenComponent aria-describedby="aria description" />
+      );
+
+      dialogFullScreenPreview()
+        .should("have.attr", "aria-describedby")
+        .and("contain", "aria description");
+    });
+
+    it("should render DialogFullScreen component with aria-label", () => {
+      CypressMountWithProviders(
+        <DialogFullScreenComponent aria-label="aria label for dialog full screen" />
+      );
+      dialogFullScreenPreview()
+        .should("have.attr", "aria-label")
+        .and("contain", "aria label for dialog full screen");
+    });
+
+    it("should render DialogFullScreen component with aria-labelledby", () => {
+      CypressMountWithProviders(
+        <DialogFullScreenComponent aria-labelledby="33c4cb62-2e16-91f4-e939-4d2f09987d4c" />
```

Comment:
where this exactly `aria-labelledby` ?

Reply:
I thought I saw a guid for the labelledby, must have imagined it!